### PR TITLE
Add testflinger-testenv container

### PR DESCRIPTION
**Pull request by plars**
## Description

When running the test phase for agents, we currently have it run under a docker image pulled from dockerhub at plars/testflinger-testenv-focal (don't be too bothered that it's focal based, it's just to give some working env on the agent-host and this one is well tested).

Now that we're on GH for everything, we should just use ghcr.io, so that nobody has to switch it over to a new location when that bus hits me some day :)

This is the first step in resolving this. I intentionally didn't change the path pointed to by the tf-test script that the agent calls **yet**, because we need to make sure firewall rules are all adjusted and everything before rolling this out in production.

## Resolved issues
N/A

## Documentation
There's a small README there just saying what it is, but this doesn't really do anything different, just moves the location

## Web service API changes
N/A

## Tests
Successful action run: https://github.com/canonical/testflinger/actions/runs/11961073449
Published container image: https://github.com/canonical/testflinger/pkgs/container/testflinger%2Ftestflinger-testenv

I also tried pulling this container image and running it locally.

### CODE REVIEW

**Comment by plars** [comment]
A few more changes:
- Added some packages that George requested
- matrix build so that it makes a tagged version of this for focal, jammy, and noble. This way we could try out newer versions if we want.

File: .github/workflows/build-testflinger-testenv-container.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
+name: Build testflinger-testenv container image
+
+on:
+  push:
+    branches:
+      - main
+      - testflinger-testenv
```
Comment (nancyc12) [inline .github/workflows/build-testflinger-testenv-container.yml:7]: should we remove `testflinger-testenv`?

**Review by nancyc12 — DISMISSED** [review]
Thanks for making this workflow! I only have a small question :)

File: .github/workflows/build-testflinger-testenv-container.yml (hunk 1 above)
Comment (plars) [reply .github/workflows/build-testflinger-testenv-container.yml]: fixed

**Review by nancyc12 — APPROVED** [review]
LGTM+1

**Review by nancyc12 — APPROVED** [review]

**Pull request merged**